packages/config/index.js

```diff
@@ -132,19 +132,26 @@ Config.search = (options = {}, filename) => {
 };
 
 Config.detect = (options = {}, filename) => {
-  const configFile = Config.search(options, filename);
+  let configFile;
+  const configPath = options.config;
+  if (configPath) {
+    configFile = path.isAbsolute(configPath)
+      ? configPath
+      : path.resolve(configPath);
+  } else {
+    configFile = Config.search(options, filename);
+  }
 
   if (!configFile) {
     throw new TruffleError("Could not find suitable configuration file.");
   }
-
   return Config.load(configFile, options);
 };
 
 Config.load = (file, options) => {
   const config = new Config();
 
-  config.working_directory = path.dirname(path.resolve(file));
+  config.working_directory = process.cwd();
```

Comment:
I think it makes sense to do a boolean check in here or in detect to see if someone is using the config flag.

If they are, use cwd, otherwise use the current behavior.